### PR TITLE
Improve Activity operation text

### PR DESCRIPTION
**Pull request by philipwhiuk**
First commit:

* Set account name when syncing headers
* Use headerFolderName to create `displayName` to show in the text when relevant
* Refactor the generation of the text into smaller chunks
* Add tests that check the correct English activity string is used

Second commit:

* Change %s to %d for some integers being passed in for `folder_progress` - fixes an error showing up in Android Studio.

### CODE REVIEW

**Comment by philipwhiuk** [comment]
This fixes #1685

File: k9mail/src/test/java/com/fsck/k9/activity/ActivityListenerTest.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    }
+
+    @Test
+    public void synchronizeMailboxProgress_shouldResultInValidStatus() {
+        activityListener.synchronizeMailboxStarted(account, folder);
+        activityListener.synchronizeMailboxProgress(account, folder, count, count);
+
+        String operation = activityListener.getOperation(RuntimeEnvironment.application);
+
+        assertEquals("Poll account:folder"+"\\u0020"+"1/1", operation);
```
Comment (cketti) [inline k9mail/src/test/java/com/fsck/k9/activity/ActivityListenerTest.java]: Looks like this is fixed in Robolectric 3.1.1 and no tests break with the new version. Please feel free to upgrade us.

File: k9mail/src/test/java/com/fsck/k9/activity/ActivityListenerTest.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        when(account.getDescription()).thenReturn("account");
+        folder = "folder";
+        errorMessage = "errorMessage";
+        message = mock(Message.class);
+        count = 1;
+        activityListener = new ActivityListener();
+    }
+
+    @Test
+    public void folderStatusChanged__shouldResultInValidStatus() {
```
Comment (cketti) [inline k9mail/src/test/java/com/fsck/k9/activity/ActivityListenerTest.java]: You're testing `getOperation()`. So `getOperation_afterFolderStatusChanged_shouldResultInValidStatus` is probably a better name. Personally, I'd just drop the last part as it basically says "should work as expected", which is implied when writing tests.

Also, I don't like the double underscore. It's bad enough we have to use underscores for readability. Having two of them is unnecessary, though. /cc @Valodim

File: k9mail/src/test/java/com/fsck/k9/activity/ActivityListenerTest.java [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+    @Test
+    public void synchronizeMailboxFailed_shouldResultInValidStatus() {
+        activityListener.synchronizeMailboxStarted(account, folder);
+        activityListener.synchronizeMailboxFailed(account, folder, errorMessage);
+
+        String operation = activityListener.getOperation(RuntimeEnvironment.application);
+
+        assertNotEquals("Poll account:folder", operation);
+        assertNotNull(operation);
```
Comment (cketti) [inline k9mail/src/test/java/com/fsck/k9/activity/ActivityListenerTest.java]: Please check what the output should be, not what it should not be.

File: k9mail/src/test/java/com/fsck/k9/activity/ActivityListenerTest.java [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    }
+
+    @Test
+    public void synchronizeMailboxFinished__shouldResultInValidStatus() {
+        activityListener.synchronizeMailboxStarted(account, folder);
+        activityListener.synchronizeMailboxFinished(account, folder, count, count);
+
+        String operation = activityListener.getOperation(RuntimeEnvironment.application);
+
+        assertNotNull(operation);
```
Comment (cketti) [inline k9mail/src/test/java/com/fsck/k9/activity/ActivityListenerTest.java]: Same as above. This is a very weak assertion.

File: k9mail/src/main/res/values-ca/strings.xml [hunk 5]
```diff
@@ -47,7 +47,7 @@
   <string name="status_loading_account_folder_headers">Recollint capçaleres <xliff:g id="account">%s</xliff:g>:<xliff:g id="folder">%s</xliff:g><xliff:g id="progress">%s</xliff:g></string>
   <string name="status_sending_account">Enviant <xliff:g id="account">%s</xliff:g><xliff:g id="progress">%s</xliff:g></string>
   <string name="status_processing_account">Proc <xliff:g id="account">%s</xliff:g>:<xliff:g id="command">%s</xliff:g><xliff:g id="progress">%s</xliff:g></string>
-  <string name="folder_progress">\u0020<xliff:g id="completed">%s</xliff:g>/<xliff:g id="total">%s</xliff:g></string>
+  <string name="folder_progress">\u0020<xliff:g id="completed">%d</xliff:g>/<xliff:g id="total">%d</xliff:g></string>
```
Comment (cketti) [inline k9mail/src/main/res/values-ca/strings.xml]: Changes to everything but `values/strings.xml` will be lost when we pull translations from Transifex. My suggestion is to leave translated strings alone when possible or to remove them when they cause lint errors.

File: k9mail/src/main/java/com/fsck/k9/activity/ActivityListener.java [hunk 6]
```diff
@@ -31,54 +31,64 @@ public void onReceive(Context context, Intent intent) {
...
         }
+    }
+
+    private String getActionInProgressOperation(Context context) {
+        String progress = (mFolderTotal > 0 ?
+                context.getString(R.string.folder_progress, mFolderCompleted, mFolderTotal) : "");
+
+        if (mLoadingFolderName != null || mLoadingHeaderFolderName != null) {
+            String displayName = null;
+            if(mLoadingHeaderFolderName != null) {
```
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/activity/ActivityListener.java]: Missing space between `if` and `(`.

File: k9mail/src/main/java/com/fsck/k9/activity/ActivityListener.java [hunk 7]
```diff
@@ -31,54 +31,64 @@ public void onReceive(Context context, Intent intent) {
...
+                displayName = mLoadingHeaderFolderName;
+            } else if(mLoadingFolderName != null) {
```
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/activity/ActivityListener.java]: Missing space.

**Comment by philipwhiuk** [comment]
I think I fixed all the outstanding issues you raised @cketti

**Comment by cketti** [comment]
Cleaned up the tests a little bit further. I tried to include explanations for the changes:  e511b8b7c1703dd4e37e890fe95b0cffbdb92b32

**Pull request merged**